src/zarr/api/asynchronous.py

```diff
@@ -247,7 +247,10 @@ async def open(
 
     try:
         return await open_array(store=store_path, zarr_format=zarr_format, **kwargs)
-    except KeyError:
+    except (KeyError, ValueError):
+        # KeyError for a missing key
+        # ValueError for failing to parse node metadata as an array when it's
```

Comment:
maybe we should create a specific exception for this case, since "valid document but the wrong node type" is different from "invalid document"